### PR TITLE
reads socket client source only when its served

### PR DESCRIPTION
**Pull request by heldr**

### CODE REVIEW

**Comment by nkzawa** [comment]
Thanks. but `read()` is a synchronous method, thus this could be performance problem.
Additionally, we'd like to read the source only once, even if `serveClient` was called several times.

**Comment by heldr** [comment]
You're right!

**Comment by palavrov** [comment]
Current implementation makes troubles with `broserify` used in `nexe` - see jaredallard/nexe#219 and jaredallard/nexe#213
I.e. it must not read the client source when it is not served. Any ideas?

File: lib/index.js [hunk 1]
```diff
@@ -94,6 +88,14 @@ Server.prototype.checkRequest = function(req, fn) {
 Server.prototype.serveClient = function(v){
   if (!arguments.length) return this._serveClient;
   this._serveClient = v;
+
+  /**
+   * Socket.IO client source.
+   */
+  if (v) {
```
Comment (palavrov) [inline lib/index.js]: To read the file just once per instatiation add `&& !this._clientSource` i.e.
```JS
if (v && !this._clientSource)
```

**Pull request closed without merging**